aws-sdk-core/lib/aws-sdk-core/signers/v4.rb

```diff
@@ -39,10 +39,16 @@ def self.sign(context)
       #   the endpoint prefix.
       # @param [String] region The region (e.g. 'us-west-1') the request
       #   will be made to.
-      def initialize(credentials, service_name, region)
+      # @param [Array] whitelist headers provided to escape blacklist check
+      def initialize(credentials, service_name, region, whitelist_headers = [])
         @service_name = service_name
         @credentials = credentials.credentials
         @region = EndpointProvider.signing_region(region, service_name)
+        if whitelist_headers.any? && (BLACKLIST_HEADERS & whitelist_headers).any?
```

Comment:
That's true, I was thinking checking the intersection of  the 2 array before getting the result. It seems redundant, I'll remove that.